lib/EPub.php

```diff
@@ -0,0 +1,843 @@
+    {
+        // set cover
+        if ($path !== false) {
+            // remove current pointer
+            $nodes = $this->xpath->query('//opf:metadata/opf:meta[@name="cover"]');
+            foreach ($nodes as $node) {
+                $node->delete();
+            }
+            // remove previous manifest entries if they where made by us
+            $nodes = $this->xpath->query('//opf:manifest/opf:item[@id="php-epub-meta-cover"]');
+            foreach ($nodes as $node) {
+                $node->delete();
+            }
+
+            if ($path) {
+                // add pointer
+                $parent = $this->xpath->query('//opf:metadata')->item(0);
+                $node = $parent->newChild('opf:meta');
+                $node->attr('opf:name', 'cover');
+                $node->attr('opf:content', 'php-epub-meta-cover');
+
+                // add manifest
+                $parent = $this->xpath->query('//opf:manifest')->item(0);
+                $node = $parent->newChild('opf:item');
+                $node->attr('id', 'php-epub-meta-cover');
+                $node->attr('opf:href', 'php-epub-meta-cover.img');
+                $node->attr('opf:media-type', $mime);
+
+                // remember path for save action
+                $this->imagetoadd = $path;
+            }
+
+            $this->reparse();
+        }
+
+        // load cover
+        $nodes = $this->xpath->query('//opf:metadata/opf:meta[@name="cover"]');
+        if (!$nodes->length) {
+            return $this->no_cover();
+        }
+        $coverid = (String) $nodes->item(0)->attr('opf:content');
+        if (!$coverid) {
+            return $this->no_cover();
+        }
+
+        $nodes = $this->xpath->query('//opf:manifest/opf:item[@id="' . $coverid . '"]');
+        if (!$nodes->length) {
+            return $this->no_cover();
+        }
+        $mime = $nodes->item(0)->attr('opf:media-type');
+        $path = $nodes->item(0)->attr('opf:href');
+        $path = dirname('/' . $this->meta) . '/' . $path; // image path is relative to meta file
+        $path = ltrim($path, '/');
+
+        $zip = new ZipArchive();
+        if (!@$zip->open($this->file)) {
+            throw new Exception('Failed to read epub file');
+        }
+        $data = $zip->getFromName($path);
+
+        return array(
+            'mime'  => $mime,
+            'data'  => $data,
+            'found' => $path
+        );
+    }
+
+    public function getCoverItem()
+    {
+        $nodes = $this->xpath->query('//opf:metadata/opf:meta[@name="cover"]');
+        if (!$nodes->length) {
+            return NULL;
+        }
+
+        $coverid = (String)$nodes->item(0)->attr('opf:content');
+        if (!$coverid) {
+            return NULL;
+        }
+
+        $nodes = $this->xpath->query('//opf:manifest/opf:item[@id="' . $coverid . '"]');
+        if (!$nodes->length) {
+            return NULL;
+        }
+
+        return $nodes->item(0);
+    }
+
+    public function Combine($a, $b)
```

Comment:
can you explain what this function is supposed to do? This should definitely have some unit tests